internal/abi/fastly/hostcalls_guest.go

```diff
@@ -65,27 +65,26 @@ func fastlyABIInit(abiVersion prim.U64) FastlyStatus
 //	   (result $err $fastly_status)
 //	 )
 //
-//go:wasm-module fastly_uap
-//export parse
+//go:wasmimport fastly_uap parse
 //go:noescape
 func fastlyUAPParse(
-	userAgent prim.Wstring,
+	userAgentData uint32, userAgentLen uint32,
 
-	family *prim.Char8,
+	family uint32,
```

Comment:
Could we define our own pointer type here? `prim.Pointer` maybe?  I'd feel better about that than having just `uint32` all over the place.

Reply:
That makes sense.  I was just trying to get this working, but now that I have the conversion script working I can tweak exactly what we're generating here.